modules/images/webp-uploads/load.php

```diff
@@ -96,6 +96,32 @@ function webp_uploads_create_sources_property( array $metadata, $attachment_id )
 			continue;
 		}
 
+		$original_image_filesize = isset( $metadata['filesize'] ) ? (int) $metadata['filesize'] : 0;
+		$webp_image_filesize     = isset( $image['filesize'] ) ? (int) $image['filesize'] : 0;
+		if ( $original_image_filesize > 0 && $webp_image_filesize > 0 ) {
+			/**
+			 * Filter whether WebP images that are larger than the matching JPEG should be discarded.
+			 *
+			 * By default the performance lab plugin will use the mime type with the smaller filesize
+			 * rather than defaulting to `webp`.
+			 *
+			 * @since n.e.x.t
+			 *
+			 * @param bool $preferred_filesize Prioritize file size over mime type. Default true.
+			 */
+			if (
+				apply_filters( 'webp_uploads_discard_larger_generated_images', true )
+				&& $webp_image_filesize >= $original_image_filesize
+			) {
+				if ( ! file_exists( $destination ) ) {
+					continue;
+				}
```

Comment:
Is this check really needed? When would the file not exist here? I would think this is impossible to get to because if generating the file failed, we would have already bailed earlier above in the `is_wp_error` check.

Reply:
Thanks for pointing it out. Initially, I didn't use it, but after feedback from someone, I included it for verification before deleting it. You are right, we don't need to check it. Changes were made to the PR.